File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/SEMINAR/S1/untitled/Sem1-MAP/src/StrategyTaskRunner.java
public class StrategyTaskRunner implements TaskRunner{
    private Container container;

    public StrategyTaskRunner(Strategy strategy) {
        container=new TaskContainerFactory().createContainer(strategy);
    }

    @Override
    public void executeOneTask() {
        Task t=container.remove();
        t.execute();
    }

    @Override
    public void executeAll() {
        while(hasTask()){
            executeOneTask();
        }
    }

    @Override
    public void addTask(Task t) {
        container.add(t);
    }

    @Override
    public boolean hasTask() {
        return !container.isEmpty();
    }
}

File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 11 - homework/lab11/lab11/task/StrategyTaskRunner.cs
namespace lab11;

public class StrategyTaskRunner:TaskRunner
{
    private Container container;

    public StrategyTaskRunner(Strategy strategy)
    {
        if (strategy == Strategy.Fifo)
        {
            container = new QueueContainer(10);
        }
        else
        {
            container = new StackContainer(10);
        }
    }
    public void ExecuteOneTask()
    {
        container.Remove().Execute();
    }

    public void ExecuteAll()
    {
        var size = container.Size();
        for (var i = 0; i < size; i++)
        {
            container.Remove().Execute();
        }
    }

    public void AddTask(Task task)
    {
        container.Add(task);
    }

    public bool HasTasks()
    {
        return container.Size() < 1;
    }
}